test/production/app-dir-edge-runtime-with-wasm/index.test.ts

```diff
@@ -0,0 +1,62 @@
+import path from 'path'
+import { createNext, FileRef } from 'e2e-utils'
+import { NextInstance } from 'test/lib/next-modes/base'
+import { renderViaHTTP } from 'next-test-utils'
+
+const files = {
+  'app/layout.tsx': `
+    export default function AppLayout({ children }) {
+      return (
+        <html>
+          <head>
+            <title>WASM Import</title>
+          </head>
+          <body>
+            {children}
+          </body>
+        </html>
+      )
+    }
+  `,
+  'app/page.tsx': `
+    // @ts-expect-error
+    import wasm from '../wasm/add.wasm?module'
+
+    console.log(wasm)
+
+    export default function Page() {
+      return "index page"
+    }
+
+    export const runtime = "experimental-edge"
+  `,
+  'wasm/add.wasm': new FileRef(path.join(__dirname, 'add.wasm')),
+}
+
+describe('app-dir edge runtime with wasm', () => {
+  let next: NextInstance
+
+  beforeAll(async () => {
+    next = await createNext({
+      files,
+      dependencies: {
+        react: 'experimental',
+        'react-dom': 'experimental',
+        typescript: 'latest',
+        '@types/react': 'latest',
+        '@types/node': 'latest',
+      },
+      nextConfig: {
+        experimental: {
+          appDir: true,
+        },
+      },
+    })
+  })
+  afterAll(() => next.destroy())
+
+  it('should have built', async () => {
+    const html = await renderViaHTTP(next.url, '/')
+    expect(html).toContain('index page')
+  })
```

Comment:
Can we execute the wasm? & see that it actually loads the Wasm file correctly in runtime? <picture data-single-emoji=":pray_parrot:" title=":pray_parrot:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..W6KHXrTJYtDDr7fR.c8Mbhf9-JnC95HxR2PE9KB3Opa1NWtHOPRwhxqvoyXQoly4mxKZ7CopH13f3AEha93sVOz18uY4lk6omWfce0pY1l04ikSZFWSlwiVlTtoW_eHsfrfN1jBm-xQ8.NZ9ZEW43fGtYw2M0Cf2t2Q" alt=":pray_parrot:" align="absmiddle"></picture>